lib/metadata/bucketfile/backend.js

```
@@ -29,152 +21,41 @@ class BucketFileInterface {
-        con.pipe(this.client.createRpcStream()).pipe(con);
-        this.metastore = this.client.sublevel(METASTORE);
-    }
-
-    /**
-     * Wait for reconnect do be done
-     * @param {function} cb - callback()
-     * @return {undefined}
-     */
-    reConnect(cb) {
-        if (this.refcnt === this.waitreco) {
-            /* Either we are alone waiting for reconnect or all
-               operations are waiting for reconnect, then force
-               a reco and notify others */
-            this.realReConnect();
-            if (this.waitreco > 1) {
-                this.notifier.emit('recodone');
-            }
-            return cb();
-        }
-        /* We need to wait for somebody to wake us up either by
-           recodone or refdecr */
-        let cbDone = false;
-
-        if (this.notifier.listeners('recodone').length > 0 ||
-            this.notifier.listeners('refdecr').length > 0) {
-            setTimeout(() => {
-                this.reConnect(cb);
-            }, 50);
-            return undefined;
-        }
-        this.notifier.once('recodone', () => {
-            if (cbDone) {
-                return undefined;
-            }
-            cbDone = true;
-            return cb();
-        });
-        this.notifier.once('refdecr', () => {
-            /* We need to recheck for the condition as
-               somebody might issue a command before we
-               get the notification */
-            if (cbDone) {
-                return undefined;
-            }
-            cbDone = true;
-            process.nextTick(() => {
-                this.reConnect(cb);
+        this.metastore.put(
+            constants.usersBucket,
+            usersBucketAttr.serialize(), err => {
+                if (err) {
+                    this.logger.error('error writing usersBucket ' +
+                                      'attributes to metadata',
+                                      { error: err });
```

If this error is due to something other than the bucket already being there, we probably do not want to even start up so we should throw an error.